libs/exavmlib/lib/IO.ex

```diff
@@ -0,0 +1,38 @@
+#
+# This file is part of AtomVM.
+#
+# Copyright 2024 Davide Bettio <davide@uninstall.it>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+#
+
+defmodule IO do
+  # This avoids crashing the compiler at build time
+  @compile {:autoload, false}
+
+  def puts(string) do
+    :io.put_chars([to_chardata(string), ?\n])
+  end
+
+  defp to_chardata(list) when is_list(list), do: list
```

Comment:
Is the `is_list` guard sufficient?  What happens if list is a list of things other than `0..255` ints?

Reply:
@bettio remarked in conversation that `io:put_chars` will handle failure scenarios.